src/RequestHandle.cpp

```diff
@@ -175,6 +215,11 @@ auto RequestHandle::SetVersion(
     }
 }
 
+auto RequestHandle::SetMaxBytes(size_t max_bytes) -> void
```

Comment:
this seems broken, as it doesn't call init() and thus the CURLOPT_XFERINFOFUNCTION might not be set

Reply:
Totally missed that! Expanded the function to reset all the related variables and set the corresponding write callback and CURLOPT_XFERINFOFUNCTION.